synapse/res/templates/saml_error.html

```diff
@@ -2,10 +2,17 @@
 <html lang="en">
 <head>
     <meta charset="UTF-8">
-    <title>SSO error</title>
+    <title>SSO login error</title>
 </head>
 <body>
-    <p>Oops! Something went wrong during authentication<span id="errormsg"></span>.</p>
+{# a 403 means we have actively rejected their login #}
```

Comment:
I looked briefly, but did you check if we need similar changes on the fallback login page?

Reply:
I don't think any of the fallback flows are involved here. If we throw a 403 here, we never redirect back to the fallback flows.